src/client.ts

```diff
@@ -90,6 +90,22 @@ export class Client {
     );
   }
 
+  public get last_txn_time(): Date | undefined {
+    return this.#lastTxn;
+  }
+
+  public set last_txn_time(time: Date | undefined) {
+    if (this.last_txn_time != undefined && time != undefined) {
+      if (time.valueOf() < this.last_txn_time.valueOf()) {
+        throw new Error("Must be greater than current value");
+      }
+    } else if (time == undefined && this.last_txn_time != undefined) {
+      throw new Error("Unable to clear last_txn_time");
+    }
+
+    this.#lastTxn = time;
+  }
+
```

Comment:
I don't think we need to let them set this to undefined. Only to the future.

Also you don't need accessor modifiers in JS.

Additionally, let's stay consistent on snake vs camelCase in the field names of this class.

Finally, this is also the **reference driver** write good docs for all parts of the interface.

```suggestion
  /**
   * @return the last transaction time seen by this client, or undefined if this client has not seen a transaction time.
   */
  get lastTxn(): Date | undefined {
    return this.#lastTxn;
  }

   /**
   * Sets the last transaction time of this client.
   * @param lastTxn - the last transaction time to set.
   * @throws Error if lastTxn is before the current lastTxn of the driver
   */
  set lastTxnl(astTxn: Date) {
      if (this.lastTxn !== undefined && lastTxn.getTime() < this.lastTxn.getTime()) {
        throw new Error("Must be greater than current value");
      }
      this.#lastTxn = time;
  }
```